website/docs/r/lab_service_plan.html.markdown

```diff
@@ -0,0 +1,128 @@
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_lab_service_plan" "example" {
+  name                = "example-lp"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  allowed_regions     = [azurerm_resource_group.example.location]
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Lab Service Plan. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Lab Service Plan should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) The Azure Region where the Lab Service Plan should exist. Changing this forces a new resource to be created.
+
+* `allowed_regions` - (Required) The allowed regions for the lab creator to use when creating labs using this Lab Service Plan. The allowed region's count must be between `1` and `28`.
+
+* `default_auto_shutdown` - (Optional) A `default_auto_shutdown` block as defined below.
+
+* `default_connection` - (Optional) A `default_connection` block as defined below.
+
+* `default_network_subnet_id` - (Optional) The resource ID of the Subnet for the Lab Service Plan network profile.
+
+* `shared_gallery_id` - (Optional) The resource ID of the Shared Image Gallery attached to this Lab Service Plan. When saving a lab template virtual machine image it will be persisted in this gallery. The shared images from the gallery can be made available to use when creating new labs.
+
+~> **NOTE:** The built-in `Azure Lab Services` Service Principal with role needs to be assigned to the Shared Image Gallery while using this property.
+
+* `support` - (Optional) A `support` block as defined below.
+
+* `tags` - (Optional) A mapping of tags which should be assigned to the Lab Service Plan.
+
+---
+
+A `default_auto_shutdown` block supports the following:
+
+* `disconnect_delay` - (Required) The amount of time a VM will stay running after a user disconnects if this behavior is enabled. This value must be formatted as an ISO 8601 string.
+
+* `idle_delay` - (Required) The amount of time a VM will idle before it is shutdown if this behavior is enabled. This value must be formatted as an ISO 8601 string.
+
+* `no_connect_delay` - (Required) The amount of time a VM will stay running before it is shutdown if no connection is made and this behavior is enabled. This value must be formatted as an ISO 8601 string.
+
+* `shutdown_on_disconnect_enabled` - (Required) Is shutdown on disconnect enabled? Possible values are `true` and `false`.
+
+* `shutdown_when_not_connected_enabled` - (Required) Will a VM get shutdown when it hasn't been connected to after a period of time? Possible values are `true` and `false`.
+
+* `shutdown_on_idle` - (Optional) Will a VM get shutdown when it has idled for a period of time? Possible values are `LowUsage` and `UserAbsence`.
```

Comment:
`disconnect_delay`, `idle_delay` and `no_connect_delay` only come into effect when their corresponding flag is set to true i.e.`shutdown_on_disconnect_enabled`, `shutdown_when_not_connected_enabled` and `shutdown_on_idle` which seems redundant and like a bit of a janky user experience.

`disconnect_delay`, `idle_delay` and `no_connect_delay` should become Optional, and if they are set then they are enabled, if they are absent then they are disabled instead of having to specify that with an additional property.

Reply:
@stephybun , yes, you're right. For `shutdown_on_disconnect_enabled` and `shutdown_when_not_connected_enabled`, I've removed them to improve user experience. For `shutdown_on_idle`, we may not remove it since it's [enum value](https://learn.microsoft.com/en-us/rest/api/labservices/lab-plans/create-or-update?tabs=HTTP#shutdownonidlemode). 

The test result after rerun for this new update:
![image](https://user-images.githubusercontent.com/19754191/208021355-a35a5eeb-e1fb-4665-8ddf-1c4f8f1a3c04.png)